### ISSUE TITLE
Upgrade from 5.5.0 to 6.6.0 kendo-theme-bootstrap compile error Can't find stylesheet to import

### ISSUE THREAD

**Issue by molequetu** [telerik/kendo-themes#4656]
**Describe the bug**
When trying to upgrade the @progress/kendo-theme-bootstrap from 5.5.0 to 6.6.0, when compiling sass with gulp compilation and VS Task Runner Explorer errors occur with error message 

```
Error in plugin "sass"
Message:
    node_modules\@progress\kendo-theme-bootstrap\scss\core\functions\index.import.scss
Error: Can't find stylesheet to import.
  ╷
1 │ @import "@progress/kendo-theme-core/scss/functions/index.import.scss";
  │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ╵
  node_modules\@progress\kendo-theme-bootstrap\scss\core\functions\index.import.scss 1:9  @import
  node_modules\@progress\kendo-theme-bootstrap\scss\_variables.scss 1:9                   @import
  node_modules\@progress\kendo-theme-bootstrap\scss\core\_index.scss 4:9                  @import
  node_modules\@progress\kendo-theme-bootstrap\scss\typography\_index.scss 1:9            @import
[11:52:41] Finished 'sass-kendo' after 121 ms
  Content\kendo\scss\backoffice-kendo.scss 58:9                                           root stylesheet
Process terminated with code 0.
```

**To reproduce**

Using gulp to compile themes as described here https://github.com/telerik/kendo-themes/wiki/Compiling-themes#using-gulp with gulp-sass v5 and dart-sass

Gulpfile.js

```
const gulp = require('gulp');
const path = require('path');
const sass = require('gulp-sass')(require('sass'));
const postcss = require("gulp-postcss");
const autoprefixer = require("autoprefixer");
const calc = require("postcss-calc");

// An importer that redirects relative URLs starting with "~" to
// node_modules.
// It used to preprocess sass kendo theme imports
function packageImporter(url) {

    // If the doesn't start with ~, return it as is.
    if (!url.startsWith('~')) {
        return null;
    }

    const nodeModules = path.resolve('./node_modules');

    // Create a new url to the correct location
    const file = path.resolve(
        nodeModules,
        url.slice(1)
    );

    return { file };
}

/**
 * Init options for postcss 
 */
const postcssPlugins = [
    calc({
        precision: 10
    }),
    //https://github.com/postcss/autoprefixer
    // adds browser support 
    autoprefixer({
        overrideBrowserslist: ['> 10%']
    })
];

/**
 * Init options for sass
 * compilation
 */
const sassOptions = {
    precision: 10,
    outputStyle: 'expanded',
    importer: [packageImporter]
};


/**
 * Task to preprocess kendo SCSS theme
 */
gulp.task('sass-kendo', function () {
    return gulp.src('./Content/kendo/scss/backoffice-kendo.scss')
        .pipe(sass.sync(sassOptions).on('error', sass.logError))
        .pipe(postcss(postcssPlugins))
        .pipe(gulp.dest('./wwwroot/css'));
});
```

package.json

```
"dependencies": {
    "@progress/kendo-theme-bootstrap": "6.6.0",
    "bootstrap4": "npm:bootstrap@^4.3.1"
  },
  "devDependencies": {
    "autoprefixer": "10.4.8",
    "gulp": "4.0.2",
    "gulp-postcss": "9.0.1",
    "gulp-sass": "5.1.0",
    "postcss-calc": "8.2.4",
    "sass": "1.54.2"
  }
```
My main kendo .scss file is using imports like this:

```
// Treeview   
$kendo-treeview-item-focus-shadow: inset 0 0 0 2px $primary;

@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/grid/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/treelist/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/combobox/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/numerictextbox/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/multiselect/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/dropdownlist/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/datetimepicker/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/datepicker/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/timepicker/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/maskedtextbox/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/listbox/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/editor/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/dialog/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/window/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/tooltip/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/tabstrip/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "~@progress/kendo-theme-bootstrap/scss/treeview/_index.scss";

// custom overrides, add in the end for CSS specificity
@import "_custom.scss";
```

Running 

**Expected behavior**
Theme should compile and not fail with cannot find imports, ideally in every upgrade of package.

**Affected package (please remove the unneeded items)**
* kendo-theme-bootstrap

**Affected suites (please remove the unneeded items)**
* Kendo UI for jQuery - ASPNET MVC

**Affected browsers (please remove the unneeded items)**
* Not Applicable

**Build system information (please remove the unneeded items)**
* OS: ,Windows
* Node version: 12
* NPM version:  6.14

**Additional context**
As I understand there is an issue again with paths and relative paths and the use of ~ tilde which was in earlier versions. checking new source files from 6.6.0 theme are not using tilde and the error occurs there.

**Comment by molequetu** [comment]
@joneff  any insights on this; 

Are there change after 6.0.0 release of sass kendo-theme-bootstrap regarding the structure of importing partially files;

**Comment by zhpenkov** [comment]
Hello @molequetu,

We're pleased to inform you that we've taken your concern and have promptly addressed it. The wiki page has been updated to incorporate the necessary changes. You can now review the updated information to resolve the encountered issue.
https://github.com/telerik/kendo-themes/wiki/Compiling-themes

With the **6.0.0** release, we decided to deprecate tilde imports in the themes. It is no longer needed to use the **"~"** symbol before an import in the **.scss** file. To ensure functionality, the removal of the prefixes in the imports is required.

```
@import "~@progress/kendo-theme-*/scss" //invalid
@import "@progress/kendo-theme-*/scss" //valid
```
To reflect the depreciation of the tilde prefix, we have eliminated the **packageImporter** function as it is now unnecessary. 
We have utilized **includePaths** in the **sassOptions**. More about **includePaths** can be read here:
https://sass-lang.com/documentation/js-api/interfaces/legacyfileoptions/#includePaths

To improve your compile time performance, it's advisable to use 'sass-embedded' and minimize the number of repeated file imports **("--/scss/typography/_index.scss",  "--/scss/utils/_index.scss")** in the main .scss file.

**Comment by molequetu** [comment]
@zhpenkov thank you for your response and proposed solution in order to compile the > 6v themes.

I managed to compiled them with the solution you suggest, removing the ~ tidle symbol from my imports and not using furthermore packageImporter function but sass includePaths option.

Although, the themes compilation time went much bigger(~4m) compared to previous compilation time() 

I will now try to use sass-embedded as you suggest and check the difference

Regarding the number of repeated file imports, my main.scss was exported by ThemeBuilder so all these imports come from the ThemeBuilder (previous version and not PRO that is now) for the components we use.

Can you help me with which of these imports are redundant and can remove?

**Comment by zhpenkov** [comment]
Hello, @molequetu,

Sure, I'd be happy to help! In order to identify redundant imports, we must check which ones are used twice or more. Including desired imports only once in the **SCSS** file will greatly reduce the amount of time it takes to compile.

Here is an example of how the imports in your current main **SCSS** file should appear:
```scss
@import "@progress/kendo-theme-bootstrap/scss/typography/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/utils/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/grid/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/treelist/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/combobox/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/numerictextbox/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/multiselect/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/dropdownlist/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/datetimepicker/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/datepicker/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/timepicker/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/maskedtextbox/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/listbox/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/editor/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/dialog/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/window/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/tooltip/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/tabstrip/_index.scss";
@import "@progress/kendo-theme-bootstrap/scss/treeview/_index.scss";
``` 
The duplicate imports of `@import "@progress/kendo-theme-bootstrap/scss/typography/_index.scss";` and `@import "@progress/kendo-theme-bootstrap/scss/utils/_index.scss";`  are removed. Once included, the import's content is loaded into the SCSS file and there is no need for further inclusions - that will only result in decreased compile-time performance.

This suggested change will reduce the amount of time it takes to compile.
